pkg/graph/drive.resolvers.go

```diff
@@ -619,6 +837,59 @@ func (r *queryResolver) Files(ctx context.Context, input *model.FilesFilterInput
 	}, nil
 }
 
+// Shares is the resolver for the shares field.
+func (r *queryResolver) Shares(ctx context.Context, input *model.ShareInput) ([]*model.ShareUser, error) {
+	_, err := middleware.GetUser(ctx)
+	if err != nil {
+		return nil, nil
+	}
+
+	var shares []*db.Share
+
+	if input.FileID != nil {
+		err = r.DB.NewSelect().Model(&shares).Where("file_id = ?", input.FileID).Scan(ctx)
+	} else if input.BucketID != nil {
+		err = r.DB.NewSelect().Model(&shares).Where("bucket_id = ?", input.BucketID).Scan(ctx)
+	} else {
+		return nil, fmt.Errorf("file_id or bucket_id must be provided")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Make slice of user IDs
+	var userIDs []string
+	for _, share := range shares {
+		userIDs = append(userIDs, share.SharedWith)
+	}
```

Comment:
I don't think we need to do that. Having this within a sql query should be more secure and faster